etna/ciim/utils.py

```diff
@@ -14,6 +15,31 @@ def underscore_to_camelcase(word, lower_first=True):
     return result
 
 
+def extract(source: Any, key: str, default: Any = "", fail_loudly: bool = False) -> Any:
+    """
+    Utilises Django template engine's variable resolution logic to extract
+    values from unpredictable nested data structures using a simple '.'
+    separated string to indicate keys, attributes, methods and indexes to be
+    traversed or called.
+    """
+    variable_str = f"source.{key}"
+    try:
+        # attempt regular initialization to trigger validation
+        variable_node = Variable(variable_str)
+    except TemplateSyntaxError:
```

Comment:
Can we have tests around `TemplateSyntaxError`?

Reply:
Yes

Reply:
This is irrelevant now, as `VariableDoesNotExist` is not used